app/workers/stale_issue_marker.rb

```diff
@@ -0,0 +1,72 @@
+class StaleIssueMarker
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot_glacial, :retry => false
+
+  include SidekiqWorkerMixin
+
+  # If an issue/pr has any of these labels, it will not be marked as stale or closed
+  PINNED_LABELS    = ['pinned']
+
+  STALE_LABEL_NAME = 'stale'
+  STALE_DATE       = 3.months.ago
+  STALE_ISSUE_MESSAGE = <<-EOS
+This issue has been automatically marked as stale because it has not been updated for at least 3 months.
+
+If you can still reproduce this issue on the current release or on `master`, please reply with all of the information you have about it in order to keep the issue open.
+
+Thank you for all your contributions!
+  EOS
+  CLOSABLE_PR_MESSAGE = <<-EOS
+This pull request has been automatically closed because it has not been updated for at least 3 months.
+
+Feel free to open a new pull request if this work is still valid.
+
+Thank you for all your contributions!
+  EOS
+
+  attr_reader :fq_repo_name
+
+  def perform(fq_repo_name)
+    @fq_repo_name = fq_repo_name
+
+    GithubService.issues(fq_repo_name, :state => :open, :sort => :updated, :direction => :asc).select do |issue|
+      pinned  = (issue.labels & PINNED_LABELS).present?
+      stale_issue = issue.updated_at < STALE_DATE && !issue.pull_request? && !pinned
+      closable_pr = issue.updated_at < STALE_DATE && issue.pull_request? && !pinned
+
+      if stale_issue
+        stale_issues << issue
+      elsif closable_pr
+        closable_prs << issue
+      end
+    end
+
+    if GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)
+      stale_issues.each do |issue|
+        unless issue.labels.include?(STALE_LABEL_NAME)
+          Rails.logger.info("[#{Time.now}] - Marking issue #{fq_repo_name}##{issue.number} as stale")
+          issue.add_labels([STALE_LABEL_NAME])
+          issue.add_comment(STALE_ISSUE_MESSAGE)
+        end
+      end
+    else
+      raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}"
```

Comment:
Prefer moving the raise up instead of buried in the else condition:

```ruby
raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}" unless GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)

stale_issues.each do |issue|
  #...
end